xbmc/filesystem/SMBDirectory.cpp

```diff
@@ -31,6 +31,7 @@
 
 #include "system.h"
 
+#ifdef HAS_FILESYSTEM_SMB
```

Comment:
building with -DENABLE_SMBCLIENT=OFF should be enough. see https://github.com/xbmc/xbmc/blob/master/xbmc/filesystem/CMakeLists.txt#L160-L165